pandas/_libs/tslibs/parsing.pyx

```diff
@@ -904,6 +909,22 @@ def guess_datetime_format(
     #  that any user-provided function will not either.
     tokens = dt_str_split(dt_str)
 
+    # Normalize timezone tokens
+    if (parsed_datetime.tzinfo is not None) and len(tokens) > 1:
+        if tokens[-1] == "Z":
+            # the last "Z" means zero offset
+            tokens[-1] = "+0000"
+        else:
+            # If the input string has a timezone offset like '+0900',
```

Comment:
What if there's a separator like `+09:00`?

Reply:
Thanks again.

This`guess_datetime_format` fails to parse offsets including ":" separator and returns None.
In terms of completeness of the function `guess_datetime_format`, I think that offsets with separators should be parsed correctly. So I will try.

Note that even in the current version, `to_datetime` can parse timezone like "+09:00" correctly. This behavior is the same as the current master.